packages/twenty-server/src/engine/core-modules/auth/services/sign-in-up.service.ts

```diff
@@ -113,19 +113,38 @@ export class SignInUpService {
       }
     }
 
-    if (workspaceInviteHash) {
-      return await this.signInUpOnExistingWorkspace({
+    const invitationValidation =
+      workspacePersonalInviteToken || workspaceInviteHash
+        ? await this.workspaceInvitationService.validateInvitation({
+            workspacePersonalInviteToken: workspacePersonalInviteToken,
+            workspaceInviteHash,
+            email,
+          })
+        : null;
```

Comment:
if (workspacePersonalInviteToken || workspaceInviteHash) {
 const invitationValidation =
    await this.workspaceInvitationService.validateInvitation({
            workspacePersonalInviteToken: workspacePersonalInviteToken,
            workspaceInviteHash,
            email,
          })

  if (
      invitationValidation?.isValid === true &&
      invitationValidation.workspace
    ) {
}
}

Reply:
not sure it's better